### PR TITLE
Adjust tests to work with secure cluster

### PR DESCRIPTION
**Pull request by Jakob3xD**
### Description
- Update workflow action versions
- Move functions from opensearchapi/internal/test to internal/test
- Adjust integration test to work with secure opensearch
- adjust golangci-lint config for internal/ folder
- add missing merge_id field for indices segement request
- change custom_foldername field to pointer as it can be null
- skip render search template test with security due to issue

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Jakob3xD** [comment]
Should be easier to review this MR commit by commit.

File: internal/test/helper.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/stretchr/testify/require"
+	"github.com/wI2L/jsondiff"
+
+	"github.com/opensearch-project/opensearch-go/v3"
+	"github.com/opensearch-project/opensearch-go/v3/opensearchapi"
+)
+
+// NewClient returns an opensearchapi.Client that is adjusted for the wanted test case
+func NewClient() (*opensearchapi.Client, error) {
+	if os.Getenv("SECURE_INTEGRATION") == "true" {
```
Comment (dblock) [inline internal/test/helper.go:29]: You should be able to refactor this away by having `ClientConfig()` return `nil` when there isn't one defined.

**Review by dblock — APPROVED** [review]
This is good and I'd merge as is. 

I think having `SECURE_INTEGRATION` being referenced as an environment variable in more than one place isn't great. I would set `OPENSEARCH_USERNAME`, `OPENSEARCH_PASSWORD`, and `OPENSEARCH_URL`m, rather than trying to drive everything off the one environment variable to make it more idiomatic, but this can be improved later (or not).

**Comment by Jakob3xD** [comment]
> I think having `SECURE_INTEGRATION` being referenced as an environment variable in more than one place isn't great.

It is now only referenced in the `IsSecure()` function. It makes more sense this way.

**Review by dblock — APPROVED** [review]
Good work!

**Pull request merged**